acbuild/abort.go

```diff
@@ -42,14 +42,27 @@ func runAbort(cmd *cobra.Command, args []string) (exit int) {
 		return 1
 	}
 
+	lockfile, err := getLock()
+	if err != nil {
+		stderr("abort: %v", err)
+		return 1
+	}
+	// Lock will be released when lib.Abort deletes the folder containing the
+	// lockfile.
+
 	if debug {
 		stderr("Aborting the build")
 	}
 
-	err := lib.Abort(path.Join(contextpath, workprefix))
+	err = lib.Abort(path.Join(contextpath, workprefix))
 
 	if err != nil {
 		stderr("abort: %v", err)
+		// In the event of an error the lockfile won't have been removed, so
+		// let's release the lock now
+		if err := releaseLock(lockfile); err != nil {
```

Comment:
This is a bit pedantic, but: it's possible for Abort() to fail, yet for it to have successfully removed the lockfile (for example, if the RemoveAll fails partway through after it has removed the lockfile), in which case the releaseLock will fail somewhat spuriously. One possible solution would be for releaseLock to not fail if its os.Remove() returns os.ErrNotExist.